enterprise_catalog/apps/api/v1/views/enterprise_customer.py

```diff
@@ -119,8 +124,38 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
         }
         if (get_catalogs_containing_specified_content_ids or get_catalog_list):
             response_data['catalog_list'] = catalogs_that_contain_course
+
+        response_data[RESTRICTED_RUNS_ALLOWED_KEY] = self._get_restricted_runs_allowed_for_query(
+            course_run_ids, customer_catalogs,
+        )
         return Response(response_data)
 
+    def _get_restricted_runs_allowed_for_query(self, course_run_ids, customer_catalogs):
+        # filter the set of restricted course keys down to only
+        # those requested by the client, and only if those requested keys
+        # are top-level course keys (NOT course run keys).
+        requested_course_keys = {
+            key for key in course_run_ids if not key.startswith(COURSE_RUN_KEY_PREFIX)
+        }
+        serialized_data = defaultdict(lambda: defaultdict(dict))
```

Comment:
I feel like taken to its logical extreme, this would simplify the loop logic:
```suggestion
        serialized_data = defaultdict(lambda: defaultdict(lambda: {'catalog_uuids': set()}))
```